homeassistant/components/neato/__init__.py

```diff
@@ -1,197 +1,117 @@
-    "ui_alert_recovering_location": "Returning to start",
-    "ui_alert_battery_chargebasecommerr": "Battery error",
-    "ui_alert_busy_charging": "Busy charging",
-    "ui_alert_charging_base": "Base charging",
-    "ui_alert_charging_power": "Charging power",
-    "ui_alert_connect_chrg_cable": "Connect charge cable",
-    "ui_alert_info_thank_you": "Thank you",
-    "ui_alert_invalid": "Invalid check app",
-    "ui_alert_old_error": "Old error",
-    "ui_alert_swupdate_fail": "Update failed",
-    "dustbin_full": "Please empty dust bin",
-    "maint_brush_change": "Change the brush",
-    "maint_filter_change": "Change the filter",
-    "clean_completed_to_start": "Cleaning completed",
-    "nav_floorplan_not_created": "No floorplan found",
-    "nav_floorplan_load_fail": "Failed to load floorplan",
-    "nav_floorplan_localization_fail": "Failed to load floorplan",
-    "clean_incomplete_to_start": "Cleaning incomplete",
-    "log_upload_failed": "Logs failed to upload",
-}
-
-
-def setup(hass, config):
+
+async def async_setup(hass, config):
     """Set up the Neato component."""
+
+    if NEATO_DOMAIN not in config:
+        # There is an entry and nothing in configuration.yaml
+        return True
+
+    entries = hass.config_entries.async_entries(NEATO_DOMAIN)
+    hass.data[NEATO_CONFIG] = config[NEATO_DOMAIN]
+
+    if entries:
+        # There is an entry and something in the configuration.yaml
+        entry = entries[0]
+        conf = config[NEATO_DOMAIN]
+        if (
+            entry.data[CONF_USERNAME] == conf[CONF_USERNAME]
+            and entry.data[CONF_PASSWORD] == conf[CONF_PASSWORD]
+            and entry.data[CONF_VENDOR] == conf[CONF_VENDOR]
+        ):
+            # The entry is not outdated
+            return True
+
+        # The entry is outdated
+        hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
+
+    # Create the new entry
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            NEATO_DOMAIN, context={"source": SOURCE_IMPORT}, data=config[NEATO_DOMAIN]
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up config entry."""
     from pybotvac import Account, Neato, Vorwerk
 
-    if config[DOMAIN][CONF_VENDOR] == "neato":
-        hass.data[NEATO_LOGIN] = NeatoHub(hass, config[DOMAIN], Account, Neato)
-    elif config[DOMAIN][CONF_VENDOR] == "vorwerk":
-        hass.data[NEATO_LOGIN] = NeatoHub(hass, config[DOMAIN], Account, Vorwerk)
+    if entry.data[CONF_VENDOR] == "neato":
+        hass.data[NEATO_LOGIN] = NeatoHub(hass, entry.data, Account, Neato)
+    elif entry.data[CONF_VENDOR] == "vorwerk":
+        hass.data[NEATO_LOGIN] = NeatoHub(hass, entry.data, Account, Vorwerk)
+
     hub = hass.data[NEATO_LOGIN]
-    if not hub.login():
+    if not hub.logged_in:
         _LOGGER.debug("Failed to login to Neato API")
         return False
+
     hub.update_robots()
```

Comment:
This is doing I/O inside an async function and should be run in the executor.

Reply:
done